src/serverRenderer.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import { StaticRouter } from 'react-router-dom';
+import { matchRoutes } from 'react-router-config';
+import Root from './Root';
+import SearchPage from './components/SearchPage';
+import MoviePage from './components/MoviePage';
+import NotFoundPage from "./components/NotFoundPage";
+import configureStore from './actions/store';
+
+const routes = [
+  { path: '/', exact: true, component: SearchPage },
+  { path: '/search/:query', exact: true, component: SearchPage },
+  { path: '/film/:id', exact: true, component: MoviePage },
+  { path: '/*', component: NotFoundPage }
```

Comment:
Routes configuration is now here and in `Root` component. Try removing this duplication.